docs/units/distribution_stretchgoal_api.rst

```diff
@@ -0,0 +1,57 @@
+.. doctest-skip-all
+>>> # code requires imports and init from distribution_api.rst
+
+"Stretch goals" (i.e., in the plan but could be done later):
+
+Have the named/analytic distributions actually propagate analytically when it makes sense, instead of Monte Carlo:
+
+>>> real_distr1 = u.NormalDistribution(10*u.kpc, std=3*u.kpc, n_samples=None)
+>>> real_distr2 = u.NormalDistribution(10*u.kpc, std=4*u.kpc, n_samples=None)
+>>> distr12_plus = real_distr1 + real_distr2
+>>> distr12_plus.q
+<Quantity 20.0 kpc>
+>>> distr12_plus.unc  # analytically should be the quadrature sum of the components
+<Quantity 5.0 kpc>
+
+>>> distr12_mul = real_distr1 * real_distr2
+>>> distr12_mul.pdf_
+<Quantity 19.79716280988112 kpc>
+>>> distr12_plus.pdf_std  # analytically should be the quadrature sum of the *pecentage* uncertainty of the components
+<Quantity 4.196109517371489 kpc>

```

Comment:
We need to be careful here that we still add the underlying samples and not re-generate random numbers, in order to take into account correlated values. For instance, doing ``(d1 + d2)/d1`` should be exactly equivalent to ``1 + d2 / d1``. In general I'm not a fan of mixing the analytical with the Monte-Carlo stuff for that reason.

For instance, should ``pdf_std`` ever be the input analytical value, or should it *always* be the actual ``std`` of the underlying sample?

Reply:
As currently implemented everything is Monte Carlo, so this shouldn't present an issue.  In the commits I just pushed up I said explcitly that it's all sampled but that we reserve the *possibility* of analytic representations in the future - so kicking that can down the road to be solved only if necessary.